vault_cli/cli.py

```diff
@@ -183,18 +183,28 @@ def list_(client_obj: client.VaultClientBase, path: str):
 
 
 @cli.command(name="get-all")
+@click.option(
+    "--flat",
+    is_flag=True,
+    help=("Returns the full path as keys instead of merging paths into a tree"),
+)
 @click.argument("path", required=False, nargs=-1)
 @click.pass_obj
 @handle_errors()
-def get_all(client_obj: client.VaultClientBase, path: Sequence[str]):
+def get_all(client_obj: client.VaultClientBase, path: Sequence[str], flat: bool):
     """
     Return multiple secrets. Return a single yaml with all the secrets located
     at the given paths. Folders are recursively explored. Without a path,
     explores all the vault.
     """
     paths = list(path) or [""]
 
-    result = client_obj.get_all_secrets(*paths)
+    if flat:
+        result: types.JSONDict = {}
+        for path in paths:
+            result.update(client_obj.get_secrets(path))
```

Comment:
Hmmm, why not take the opportunity to reorganize the mess of get_all_secrets / get_secrets ?

get_all_secrets would be responsible of making multiple calls to get_secrets and have a "nested" parameter that would transform the result, or not

Reply:
moved.